PluginCode/src/main/java/smr/cs/ualberta/libcomp/DatabaseAccess.java

```diff
@@ -481,16 +484,67 @@ public void updateUserProfile(User userRecord) throws IOException {
 
         if (Integer.parseInt(userRecord.getCloudStore()) == 1) {
             sendUser(userRecord.getUserID(), obj.toString());
+
+            sendFeedbackToCloud(); // check if the user has local feedback stored previously, send them now.
         }
     }
 
-    public void updateFeedback(ReplacementFeedback replacementFeedback) throws IOException {
+    public void storeLocalFeedback(JSONObject obj) throws IOException {
+
+        String filePath = this.filePath +"\\Feedback.json";
+        File myFile = new File(filePath);
+        JSONObject Mainobj = null; // = new JSONObject();
+        if (myFile.exists()) {
+            String content = FileUtils.readFileToString(myFile, "utf-8");
+            JSONObject objM = new JSONObject(content);
+            JSONArray jsonarr = objM.getJSONArray("Feedbacks");
+            jsonarr.put(obj);
+            Mainobj = new JSONObject();
+            Mainobj.put("Feedbacks", jsonarr);
+        } else
+        {
+            Mainobj = new JSONObject();
+            JSONArray array = new JSONArray();
+            array.put(obj);
```

Comment:
What is `array`? again, descriptive name

Reply:
renamed to feedbackEntryArray